src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpHttpTraceExporter.cs

```diff
@@ -0,0 +1,102 @@
+// <copyright file="OtlpHttpTraceExporter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Google.Protobuf;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
+using OtlpCollector = Opentelemetry.Proto.Collector.Trace.V1;
+
+namespace OpenTelemetry.Exporter
+{
+    /// <summary>
+    /// Exporter consuming <see cref="Activity"/> and exporting the data using
+    /// the OpenTelemetry protocol (OTLP) over HTTP.
+    /// </summary>
+    public class OtlpHttpTraceExporter : BaseOtlpHttpExporter<Activity>
```

Comment:
Could we keep OtlpTraceExporter, as make the transport option:grpc vs http as something in the `OtlpExporterOptions` ?

Reply:
I think we can introduce additional abstraction for transport and keep OtlpTraceExporter. In this case some refactoring of BaseOtlpExporter and it's child classes will be required. At the moment base class BaseOtlpExporter already has some grpc specifics.

Reply:
@alanwest please share your thoughts, as he was planned to do some refactoring on OTLP Exporter, (in Metrics context though, but might be relevant here as well)

Reply:
I've thought a little about this in the past and considered two options:

1. Keep the OTLP/gRPC exporters and the OTLP/HTTP exporters in the same package
    * Pros: It would make for a simple configuration change if a consumer deployed a service and needed to change export between gRPC vs. HTTP. This is nice because in my experience folks don't always think about whether their backend supports OTLP over gRPC and/or HTTP.
    * Cons: If a consumer knows they want to use the HTTP exporter, they end up pulling down all the gRPC dependencies that are unnecessary for their use case. 
2. Separate the exporters into two packages - i.e., something like `OpenTelemetry.Exporter.OpenTelemetryProtocol` and `OpenTelemetry.Exporter.OpenTelemetryProtocolHttp`. A number of the other language SDKs have taken this approach.
    * Pros: Only the necessary dependencies are pulled down (no gRPC dependencies when no need for them)
    * Cons: If you consume the HTTP package only to discover that your backend only supports gRPC, then you have to recompile your application.

I personally prefer the first option. If we take this approach then I think we should keep all the functionality in a single class `OtlpTraceExporter`. All the gRPC specific functionality in the `BaseOtlpExporter` is not publicly exposed, so I'm not concerned about making the inner workings of the class figure out which transport mechanism to use. Also, the `OtlpExporterOptions` should be expanded with the [`Protocol` option](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options) as specified in the spec. Supported values would be `grpc` and `http/protobuf`.

Reply:
1. Code recompilation doesn't sound too dramatic to me. Another advantage of the 2nd option can be code simplicity. Up to now all other exporters like Jaeger, Zipkin, etc was rather straightforward. 
2. From other side the 2nd approach leads to code duplication in projects for gRPC, HTTP and potentially for WebSocket (https://github.com/open-telemetry/oteps/blob/main/text/0099-otlp-http.md#prior-art-and-alternatives). Since 
OTLP/HTTP uses the same protobuf payload that is used by OTLP/gRPC both exporters can be kept in OtlpTraceExporter.
3. If we decide go with the 2nd option then we start to diverge from other language SDKs. I'm not sure if it's good or bad thing... 

Speaking about implementation of the 2nd option: we can extract from TraceService and reuse exiting interface ITraceServiceClient. The only thing is to get rid of gRPC specific Metadata class in the interface. We can introduce generic ITraceServiceClient<TMetadata>.  I can do in scope of this issue.

Reply:
Question for @alanwest. 
According to the [Protocol specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md) the default value of the protocol is n/a. But we still need some default for the protocol configuration option. Can we use `grpc` as a default ? Anyway we had only grpc exporter implementation before. If we can't use any default then should we always expect that OTEL_EXPORTER_OTLP_PROTOCOL is always defined at client's side ?

Reply:
> Can we use `grpc` as a default ?

This seems ok to me. We'd have to default to `grpc` so as not to break the existing behavior.

---

My vote is go with option 1. @cijothomas do you have any concerns in keeping the gRPC and HTTP exporters in the same package?

Reply:
Hello @cijothomas and @alanwest ,

Changes made after this discussion have been pushed. Please, take a look.  

Changes in a nutshell
- new abstraction encapsulating export transport protocol has been introduced: `IExportClient<in TRequest>`
- transport protocol related code has been extracted from `OtlpTraceExporter`, `OtlpMetricsExporter `and moved to `OtlpGrpcMetricsExportClient `and `OtlpGrpcTraceExportClient`
- new implementation corresponding to `http/protobuf` has been added: `OtlpHttpTraceExportClient `
- only `OtlpTraceExporter `class is kept 
- instantiation of corresponding `IExportClient ` implementation is done inside of `OtlpTraceExporter ` according to the value of the `OtlpExporterOptions.Protocol` property (`OTEL_EXPORTER_OTLP_PROTOCOL`)
- `OtlpExporterOptions.Protocol` supports `grpc` (**used as default**!)  and `http/protobuf` values
- unit tests has been enhanced or added accordingly
- console test application has been updated accordingly

Reply:
I vote to keep gRCP & HTTP exporters in the same assembly.

Reply:
Based on SIG meeting on 9/21, we are proceeding to keep the same package for both Grpc, Http.
https://docs.google.com/document/d/1yjjD6aBcLxlRazYrawukDgrhZMObwHARJbB9glWdHj8/edit#heading=h.oyl71qhed6ck